components/vocabularies/api/src/main/java/org/phenotips/vocabulary/internal/solr/DefaultSolrVocabularyResourceManager.java

```diff
@@ -59,16 +73,75 @@
     @Inject
     private CacheManager cacheFactory;
 
+    @Inject
+    private Environment environment;
+
     @Override
     public void initialize(String vocabularyName) throws InitializationException
     {
+        // Get jars home path to where the jar resources are stored
+        Class<DefaultSolrVocabularyResourceManager> clazz = DefaultSolrVocabularyResourceManager.class;
+        String jarPath = clazz.getProtectionDomain().getCodeSource().getLocation().getPath();
```

Comment:
That is indeed the way to find the jar where a class is defined. However, you're getting the jar wher the `DefaultSolrVocabularyResourceManager` is defined, not the one for the vocabulary you're trying to initialize. If you want to get the right jar, you will have to call this from the vocabulary class, not from here, and pass the jarPath to this `initialize` method. But I don't quite like that.

An alternative is to assume a fixed set of configuration files that must be copied, and only try to copy those instead of looking inside the jar, in which case you don't care what the jar is. Something along the line of:

```

    public static final List<String> CONFIG_FILES = Arrays.asList("schema.xml", "solrconfig.xml", "protwords.txt", "stopwords.txt", "synonyms.txt");
...
        for (String file : CONFIG_FILES) {
            InputStream in = this.getClass().getResourceAsStream("/" + vocabularyName + "/conf/" + file);
            if (in == null) {
                continue;
            }
            Files.copy(in, dest.toPath().resolve(file));
        }